### PR TITLE
tikv:do retry on request with tikv failed  except caused by Cancelled.

### PR DESCRIPTION
**Pull request by AndreMouche**
Hi,all,
      The PR would do retry when request failed except caused by cancel.
       Retry is needed since when a `tikv` is not available, the leader of related region should be elected quickly. 
        This PR would do retry in the following cases  caused by the request `tikv` is down:
        * For connected request,  it would receive a  `EOF` Error.
        * For new request, it would receive 'connection refused' Error.
     @tiancaiamao @disksing PTAL

### CODE REVIEW

File: store/tikv/region_request.go [hunk 1]
```diff
@@ -14,7 +14,7 @@
 package tikv
 
 import (
-	"net"
+	"context"
```
Comment (disksing) [inline store/tikv/region_request.go]: Now we are using `golang.org/x/net/context"

Comment (tiancaiamao) [inline store/tikv/region_request.go]: Use this package:
```
	goctx "golang.org/x/net/context"
```

**Comment by disksing** [comment]
Can we use mock-tikv to construct a test case?

**Comment by AndreMouche** [comment]
@disksing @tiancaiamao @hhkbp2  PTAL

**Comment by disksing** [comment]
LGTM.

File: store/tikv/region_request.go [hunk 2]
```diff
@@ -163,10 +163,12 @@ func (s *RegionRequestSender) sendCopReqToRegion(ctx *RPCContext, req *coprocess
 func (s *RegionRequestSender) onSendFail(ctx *RPCContext, err error) error {
 	s.regionCache.OnRequestFail(ctx)
 
-	if netErr, ok := errors.Cause(err).(net.Error); ok {
-		if netErr.Temporary() {
-			err = s.bo.Backoff(boTiKVRPC, errors.Errorf("send tikv request error: %v, ctx: %s, try next peer later", err, ctx.KVCtx))
-		}
+	// Do retry on request failed except caused by Cancelled.
```
Comment (hhkbp2) [inline store/tikv/region_request.go]: Retry on request failure when it's not cancelled.

File: store/tikv/region_request.go [hunk 3]
```diff
@@ -163,10 +163,12 @@ func (s *RegionRequestSender) sendCopReqToRegion(ctx *RPCContext, req *coprocess
...
+	// When a store is not available, the leader of related region should be elected quickly.
+	// TODO: the number of retry time should be limited:since region may do not available
```
Comment (hhkbp2) [inline store/tikv/region_request.go]: region may be unavailable

**Comment by hhkbp2** [comment]
LGTM

**Pull request merged**